rum-injection-installer/installer/installer.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+)
+
+func validateInput(appID, site, clientToken, arch, agentUrl string, sessionSampleRate, sessionReplaySampleRate int) error {
+	if appID == "" || site == "" || clientToken == "" || agentUrl == "" {
+		return fmt.Errorf("appID, site, clientToken, and agentUrl are required")
+	}
+	if sessionSampleRate < 0 || sessionSampleRate > 100 {
+		return fmt.Errorf("sessionSampleRate must be between 0 and 100")
+	}
+	if sessionReplaySampleRate < 0 || sessionReplaySampleRate > 100 {
+		return fmt.Errorf("sessionReplaySampleRate must be between 0 and 100")
+	}
```

Comment:
Those fields are validated by the NGINX module when `nginx -t` is executed. I believe it is best to delegate the validation. What do you think?

Reply:
Sort of disagree, I think making sure (As much as possible) that the installer doesn't produce a wrong output is a good idea. Otherwise, I don't see much issue with repeating the validation, and do see the value in being able to go to a single place to view/update parameter requirements.

Reply:
> I think making sure (As much as possible) that the installer doesn't produce a wrong output is a good idea.

I agree, that's why the installer must run `nginx -t` to validate the configuration before "committing" the new configuration file. Since NGINX is the one making the call, why not defer the whole validation to NGINX? 

The benefit is not just repeating validation but also avoid the maintenance burden.

Reply:
Still disagree, I don't think the "maintenance burden" is relevant for some strings and integers in 6 lines of code, and lets the user realize this is an error in the arguments early in the validation rather than an error with the installer that was unable to produce a good output